### ISSUE TITLE
Cannot extract custom headers

### ISSUE THREAD

**Issue by v1rt** [naugtur/xhr#154]
I was wondering why I cannot extract headers when calling an api endpoint via cross domain. XHR package wasn't spitting out any errors on the console. I was only able to figure out the problem or show the error on the console when I used native XMLHttpRequest. Is this a bug in XHR package?

**Comment by naugtur** [comment]
Please include browser name and version.
Would you be so kind to share some code too to let me reproduce?

I think it could be a cross-domain request issue or just a misunderstanding or documentation issue. 
It's been always working for me (except IE8/9 where you need extra options) but if it's not, I want to fix it asap.

**Comment by v1rt** [comment]
It is a cross domain issue. But I was just wondering why XHR package wasn't spitting out the error. If it spitted the error, I could have fixed it quickly. Sure, here is the code

The browser is Chrome and version is Version 55.0.2883.95 (64-bit) on Mac OSX.

https://gist.github.com/v1rt/3c4d8c4680638127ed930637ceac7b95

**Comment by naugtur** [comment]
I'll look into it and see what I can do and if it can be considered a bug.
`Access-Control-Expose-Headers` is a must and as far as I remember, XMLHttpRequest was not calling the right callbacks for errors when trying to access headers failed for a cross-domain request. 

If `xhr` is to blame, it could be swallowing the exception on reading headers, but that's surprising... I remember testing that between old IEs and newer browsers.

I have some time reserved for opensource on Fridays, so I should be able to get to it before the weekend.

**Comment by v1rt** [comment]
Thanks a lot!

**Comment by naugtur** [comment]
Ok, I looked into it and the thing is - if you use XMLHttpRequest straight and call the function that gets a particular header, you trigger an error log in the console to let you know you weren't allowed.

But that's not what `response.headers[name]` is doing. `response.headers` is a plain array that we get from listing all available headers. I'd probably have to rewrite to proxy objects to be able to call a function when an array is requested. Not enough browser support for that anyway.

Avoiding errors and returning undefined is correct behavior, xhr is not swallowing any errors, it's just aiding the user in not triggering the error in the first place.

I know it's not helpful if you forget to expose a header, but that's the compromise. 

In your example, if you tried `response.rawRequest.getResponseHeader('x-name')` you would see the same error. xhr is using `getAllResponseHeaders` to create an array of everything that's available.

TL;DR: not a bug

**Comment by v1rt** [comment]
Ok, I'll try this `response.rawRequest.getResponseHeader('x-name')` and I'll let you know. Thanks!
